src/Uno.UI/UI/Xaml/Shapes/Shape.iOSmacOS.cs

```diff
@@ -53,30 +53,37 @@ private protected void Render(CGPath path)
 			Layer.AddSublayer(_shapeLayer);
 		}
 
-		private CALayer CreateLayer(CGPath path)
+		private void UpdateRender()
 		{
-			var pathLayer = new CAShapeLayer()
+			if (_shapeLayer is null)
 			{
-				Path = path,
-				StrokeColor = (Stroke as SolidColorBrush)?.ColorWithOpacity ?? Colors.Transparent,
-				LineWidth = (nfloat)ActualStrokeThickness,
-			};
+				// layer needs to be created
+				InvalidateArrange();
+				return;
+			}
+
+			SetFillAndStroke(_shapeLayer);
+		}
+
+		private void SetFillAndStroke(CAShapeLayer pathLayer)
+		{
+			RemoveSublayers();
 
 			switch (Fill)
 			{
 				case SolidColorBrush colorFill:
 					pathLayer.FillColor = colorFill.ColorWithOpacity;
 					break;
 
-				case ImageBrush imageFill when TryCreateImageBrushLayers(imageFill, GetFillMask(path), out var imageLayer):
+				case ImageBrush imageFill when TryCreateImageBrushLayers(imageFill, GetFillMask(pathLayer.Path), out var imageLayer):
 					pathLayer.FillColor = Colors.Transparent;
 					pathLayer.AddSublayer(imageLayer);
 					break;
 
-				case LinearGradientBrush gradientFill:
+				case GradientBrush gradientFill:
```

Comment:
It is really the same as `LinearGradientBrush` ?

Reply:
@jeromelaban GradientBrush is the base class of both LinearGradienBrush and RadialGradientBrush.  So by making this change we're enabling the Radial Gradient to work on shapes.